hpy/devel/include/common/hpydef.h

```diff
@@ -70,22 +73,29 @@ typedef struct {
     void *closure;
 } HPyGetSet;
 
+typedef struct {
+    const char *name;
+    HPyType_Spec typespec;
+} HPyTypeSpecDef;
+
```

Comment:
This indirection looks weird. Why do we need both `HPyType_Spec` **and** `HPyTypeSpecDef`?
And moreover, both have a `.name` which will make things even more confusing, especially if you don't use the same name in both places.

Note, this is a rhetorical question: I *know* why you had to introduce this, but I think it causes more problems than it solves, and the clang complaint is one of those. I propose one of the following:

```c
struct HPyDef {
    HPyDef_Kind kind;
    union {
        ...
        HPyType_Spec *typespec;
    };
} HPyDef;

struct HPyDef {
    HPyDef_Kind kind;
    union {
        ...
        HPyType_Spec typespec;
    };
} HPyDef;
```

The first option implies that you need to create a typespec separately, and that it can also be in another translation unit. The `HPyDef` can be generated using `HPyDef_TYPE`.

The second option removes the need to create a typespec explicitly. The user will need to write an `HPyDef` by hand, like this:

```c
HPyDef MyType {
    .kind = HPyDef_Kind_Type,
    .typespec = {
        .name = "MyType",
        ...
    }
}
```

Both have pros and cons, I have no strong preference for any of them